src/app/_mermaid_common/libs/factories/offline-table.factory.js

```diff
@@ -431,23 +437,33 @@ angular.module('mermaid.libs').factory('OfflineTable', [
           var relatedColumns = joinSchema.relatedColumns;
           var foreignKey = joinSchema.foreignKey;
           var relatedKey = joinSchema.relatedKey;
-
-          var noMatchRecord = _.zipObject(
-            relatedColumns,
-            _.map(relatedColumns, function() {
-              return null;
-            })
-          );
+          var many = joinSchema.many || false;
 
           where[relatedKey] = null;
           return _.map(records, function(obj) {
             obj[tablePrefix] = obj[tablePrefix] || {};
-            where[relatedKey] = _.get(obj, foreignKey);
-            var match = _.find(relatedRecords, where);
 
-            if (match === undefined) {
-              obj[tablePrefix] = noMatchRecord;
+            if (_.isFunction(joinSchema.relateFunction)) {
+              where = function(relatedRecord) {
+                return joinSchema.relateFunction(
```

Comment:
The `relateFunction` signature is different here, in the commented example, and in the offline service usage. Thinking maybe `joinSchema` arg should be removed here?